js/chat/chat.ts

```diff
@@ -60,13 +61,19 @@ class ChatMessage extends LightElement {
   @property() content = "...";
   @property() content_type: ContentType = "markdown";
   @property({ type: Boolean, reflect: true }) streaming = false;
+  @property() icon = "";
 
   render() {
-    const noContent = this.content.trim().length === 0;
-    const icon = noContent ? ICONS.dots_fade : ICONS.robot;
+    let msg_icon = ICONS.dots_fade;
+
+    // Show dots until we have content
+    const hasContent = this.content.trim().length !== 0;
+    if (hasContent) {
+      msg_icon = this.icon || ICONS.robot;
+    }
```

Comment:
I'd rather use the ternary here so that `icon` is set once with `const`.

```suggestion
    // Show dots until we have content
    const hasContent = this.content.trim().length > 0;
    const icon = hasContent ? this.icon || ICONS.robot : ICONS.dots_fade;
```